inconsistentStructName/inconsistentStructName.go

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package inconsistentStructName
+
+import (
+	"go/ast"
+
+	"golang.org/x/tools/go/analysis"
+)
+
+var Analyzer = &analysis.Analyzer{
+	Name: "inconsistentStructName",
+	Doc:  "check for inconsistent struct name in the methods definition",
+	Run:  run,
+}
+
+type firstVarName = struct {
+	Name string
+	Node ast.Node
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
+	firstVarNameForType := make(map[string]firstVarName)
+	for _, file := range pass.Files {
+		ast.Inspect(file, func(node ast.Node) bool {
+			switch funDecl := node.(type) {
+			case *ast.FuncDecl:
+				if funDecl.Recv != nil && len(funDecl.Recv.List) > 0 && len(funDecl.Recv.List[0].Names) > 0 {
+					field := funDecl.Recv.List[0]
+					typeIdent, ok := field.Type.(*ast.Ident)
```

Comment:
I would re-arrange the checks from StarExpr first. Then the check for *ast.Ident becomes easy.

```go
field := funDecl.Recv.List[0]
expr := field.Type
starExpr, ok := field.Type.(*ast.StarExpr)
if ok {
	expr = starExpr.X
}
typeIdent, ok := expr.(*ast.Ident)
if !ok {
	return false
}
// use typeIdent from here
```